falguard.py

```diff
@@ -69,13 +69,28 @@ def __init__(self, spec_path):
     def __call__(self, request, response, resource, params):
         self.process_resource(request, response, resource, params)
 
+    def _build_http_not_allowed_error(self, uri):
+        # Error 405 requires we provide the list of allowed methods
+        # for this URI. If None is found, then we produce a 404 error.
+        allowed_methods = [method for method in falcon.HTTP_METHODS
+                           if self._spec.get_op_for_request(method, uri)]
+
+        return (falcon.HTTPMethodNotAllowed(allowed_methods) if allowed_methods
+                else falcon.HTTPNotFound)
+
     def process_resource(self, request, response, resource, params):
         bravado_request = _BravadoRequest(request, params)
 
+        uri = request.uri_template
```

Comment:
Hmm, I will mix English and Python for this comment. Looks like you can `raise falcon.HTTPNotFound` at this point `if not uri`.

```suggestion
        uri = request.uri_template
        if not uri:
            raise falcon.HTTPNotFound
```

Reply:
In fact the whole test in unnecessary since the `process_resource` method is called after routing is done.